lib/Map/MapboxVectorTileImageryProvider.js

```diff
@@ -0,0 +1,339 @@
+//     if point not in any interior rings, feature is clicked
+function isFeatureClicked(rings, point) {
+    for (var i = 0; i < rings.length; i++) {
+        if (inside(point, rings[i])) { // Point is in an exterior ring
+            // Check whether point is in any interior rings
+            var inInteriorRing = false;
+            while (i+1 < rings.length && !isExteriorRing(rings[i+1])) {
+                i++;
+                if (!inInteriorRing && inside(point, rings[i])) {
+                    inInteriorRing = true;
+                    // Don't break. Still need to iterate over the rest of the interior rings but don't do point-in-polygon tests on those
+                }
+            }
+            // Point is in exterior ring, but not in any interior ring. Therefore point is in the feature region
+            if (!inInteriorRing) {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+
+MapboxVectorTileImageryProvider.prototype.pickFeatures = function(x, y, level, longitude, latitude) {
+    var that = this;
+    var url = this._buildImageUrl(x, y, level);
+
+    return loadArrayBuffer(url).then(function(data) {
+        var layer = new VectorTile(new Protobuf(data)).layers[that._layerName];
+        var point = [CesiumMath.toDegrees(longitude), CesiumMath.toDegrees(latitude)];
+
+        if (!defined(layer)) {
+            return []; // return empty list of features for empty tile
+        }
+
+        var features = [];
+        for (var i = 0; i < layer.length; i++) {
+            var feature = layer.feature(i);
+            if (feature.type === POLYGON_FEATURE && isFeatureClicked(feature.toGeoJSON(x, y, level).geometry.coordinates, point)) {
+                var featureInfo = new ImageryLayerFeatureInfo();
+                var uniqueId = feature.properties[that._featurePicking.uniqueIdProp];
+                var rowObject = that._featurePicking.regionRowObjects[uniqueId];
+                featureInfo.name = feature.properties[that._featurePicking.nameProp]; // Could be undefined
+                if (defined(rowObject)) {
+                    featureInfo.properties = rowObject;
+                    featureInfo.description = that._featurePicking.regionRowDescriptions[uniqueId];
+                    if (!defined(featureInfo.name)) {
+                        featureInfo.configureNameFromProperties(feature.properties);
+                    }
+                } else {
+                    featureInfo.properties = undefined;
+                    featureInfo.description = undefined;
+                }
+                featureInfo.data = {id: uniqueId};
+                features.push(featureInfo);
+                that._selectedRegion = uniqueId;
+            }
+
+        }
+
+        return features;
+    });
+};
+
+
+MapboxVectorTileImageryProvider.prototype.createHighlightImageryProvider = function(regionUniqueID) {
+    var that = this;
+    var styleFunc = function(FID)
+    {
```

Comment:
I had that fixed in my dev branch `mvt_dev_overzooming` [a month ago](https://github.com/TerriaJS/terriajs/commit/46c44d6f6c6c0465af05f65bac7867c97168464f#diff-d8af76e416b8de40b52fe4c11f7716bbL347) 😧 .

When I'm developing, I tend to have my "master" (in this case `mapbox_vt_region_provider`), then offshoot branches for different features (like `mvt_dev_overzooming`). I need to merge some of those changes before others back into master, or into other dev branches. Is there any way to facilitate that without breaking everything down into tiny commits (which would result in a whole heap of commits in the general history once merged)?